pkg/chunked/compression_linux.go

```diff
@@ -210,14 +210,13 @@ func readZstdChunkedManifest(blobStream ImageSourceSeekable, tocDigest digest.Di
 
 	decodedTarSplit := []byte{}
 	if tarSplitChunk.Offset > 0 {
-		tarSplitDigest := toc.TarSplitDigest.String()
-		// ignore the tar-split data if the digest was not specified
-		if tarSplitDigest != "" {
-			tarSplit, err := readBlob(tarSplitChunk.Length)
-			if err != nil {
-				return nil, nil, nil, 0, err
-			}
-
+		// we must consume the data to not block the producer
+		tarSplit, err := readBlob(tarSplitChunk.Length)
+		if err != nil {
+			return nil, nil, nil, 0, err
+		}
+		// but ignore it when the digest is not present
```

Comment:
```suggestion
		// but ignore it when the digest is not present , because we can’t authenticate it against tocDigest
```
or ”… when we can’t authenticate it”, to say not just what, but also why.

Reply:
fixed